ext/phar/phar.c

```diff
@@ -2318,24 +2318,17 @@ int phar_split_fname(const char *filename, size_t filename_len, char **arch, siz
  */
 int phar_open_executed_filename(char *alias, size_t alias_len, char **error) /* {{{ */
 {
-	char *fname;
-	php_stream *fp;
-	size_t fname_len;
-	zend_string *actual = NULL;
-	int ret;
-
 	if (error) {
 		*error = NULL;
 	}
 
-	fname = (char*)zend_get_executed_filename();
-	fname_len = strlen(fname);
+	zend_string *fname = zend_get_executed_filename_ex();
 
-	if (phar_open_parsed_phar(fname, fname_len, alias, alias_len, 0, REPORT_ERRORS, NULL, 0) == SUCCESS) {
+	if (phar_open_parsed_phar(ZSTR_VAL(fname), ZSTR_LEN(fname), alias, alias_len, 0, REPORT_ERRORS, NULL, 0) == SUCCESS) {
 		return SUCCESS;
 	}
 
-	if (!strcmp(fname, "[no active file]")) {
+	if (zend_string_equals_literal(fname, "[no active file]")) {
```

Comment:
This no longer makes sense, this string is only returned in `zend_get_executed_filename()`. This would now return `NULL` (and thus already segfault above).

Reply:
Ah indeed, surprised we don't have a test for that... AFAIK this only happens with ``eval()``?